modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2845,6 +2846,12 @@ def apply_select_indices(
             new_index = self.index if axis == 1 else None
         if new_columns is None:
             new_columns = self.columns if axis == 0 else None
+        if new_columns is not None and new_dtypes is not None:
+            if not new_dtypes.index.equals(new_columns):
+                # sanity check
```

Comment:
if that's a sanity check, maybe it should be
```python
assert new_dtypes.index.equals(new_columns), "error message"
```
instead of `if` then `raise ValueError`?